### PR TITLE
Update qtmacgoodies for an OSX crash fix #6930

### LINKED ISSUE #6930: owncloud on macOS crashes if I open the Settings window the second time
### Expected behaviour
The Settings window can be opened many times 

### Actual behaviour
owncloud on macOS crashes with the following backtrace:
```
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x7d1636e7480)
  * frame #0: 0x00007fff5caa8a1d libobjc.A.dylib`objc_msgSend + 29
    frame #1: 0x00000001000e9c11 owncloud`MacPreferencesWindowPrivate::displayPanel(int) + 125
    frame #2: 0x00000001000ea184 owncloud`MacPreferencesWindow::event(QEvent*) + 34
    frame #3: 0x0000000100673039 QtWidgets`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
    frame #4: 0x00000001006743b1 QtWidgets`QApplication::notify(QObject*, QEvent*) + 723
    frame #5: 0x000000010266ec94 QtCore`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 154
    frame #6: 0x00000001006a16bf QtWidgets`QWidgetPrivate::show_helper() + 369
    frame #7: 0x00000001006a20e6 QtWidgets`QWidget::setVisible(bool) + 844
    frame #8: 0x00000001000ea24f owncloud`MacPreferencesWindow::setVisible(bool) + 187
    frame #9: 0x0000000100067ffe owncloud`OCC::ownCloudGui::raiseDialog(QWidget*) + 40
    frame #10: 0x0000000102694b89 QtCore`QMetaObject::activate(QObject*, int, int, void**) + 1907
    frame #11: 0x000000010066b7b5 QtWidgets`QAction::activate(QAction::ActionEvent) + 307
    frame #12: 0x000000010268ed1c QtCore`QObject::event(QEvent*) + 124
    frame #13: 0x0000000100673039 QtWidgets`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
    frame #14: 0x00000001006743b1 QtWidgets`QApplication::notify(QObject*, QEvent*) + 723
    frame #15: 0x000000010266ec94 QtCore`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 154
    frame #16: 0x000000010266f925 QtCore`QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 435
    frame #17: 0x000000010a20ba2f libqcocoa.dylib`___lldb_unnamed_symbol660$$libqcocoa.dylib + 183
    frame #18: 0x000000010a20c18f libqcocoa.dylib`___lldb_unnamed_symbol672$$libqcocoa.dylib + 33
    frame #19: 0x00007fff30961475 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    frame #20: 0x00007fff3096141b CoreFoundation`__CFRunLoopDoSource0 + 108
    frame #21: 0x00007fff30944eb5 CoreFoundation`__CFRunLoopDoSources0 + 195
    frame #22: 0x00007fff3094445e CoreFoundation`__CFRunLoopRun + 1219
    frame #23: 0x00007fff30943d48 CoreFoundation`CFRunLoopRunSpecific + 463
    frame #24: 0x00007fff2fbdaab5 HIToolbox`RunCurrentEventLoopInMode + 293
    frame #25: 0x00007fff2fbda6f4 HIToolbox`ReceiveNextEventCommon + 371
    frame #26: 0x00007fff2fbda568 HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 64
    frame #27: 0x00007fff2de95363 AppKit`_DPSNextEvent + 997
    frame #28: 0x00007fff2de94102 AppKit`-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1362
    frame #29: 0x000000010a20ad7d libqcocoa.dylib`___lldb_unnamed_symbol652$$libqcocoa.dylib + 1123
    frame #30: 0x000000010266b7ac QtCore`QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 318
    frame #31: 0x000000010266f148 QtCore`QCoreApplication::exec() + 116
    frame #32: 0x0000000100007543 owncloud`main + 1194
    frame #33: 0x00007fff5db85ed9 libdyld.dylib`start + 1
```

### Steps to reproduce
1. Install a fresh copy of owncloud-client
2. Run it
3. Close the Connection Wizard **using the red X button of macOS**
4. Click on the tray icon and choose Settings
5. Closes the Settings window
6. Repeat step 4

### Server configuration
N/A

### Client configuration
Client version: 2.5.1

Operating system: macOS Mojave 10.14.2

OS language: zh_TW

Qt version used by client package (Linux only, see also Settings dialog): Qt 5.12.0 from MacPorts

Client package (From ownCloud or distro) (Linux only): built from source

Installation path of client: /Applications/MacPorts/owncloud.app/Contents/MacOS/owncloud

### Logs

1. Client logfile: Output of `owncloud --logwindow` or `owncloud --logfile log.txt`

```
12-18 19:36:47:663 [ info gui.application ]:	"################## ownCloud locale:[zh_TW] ui_lang:[] version:[2.5.1 (build 10818)] os:[macOS 10.14]"
12-18 19:36:47:663 [ info gui.application ]:	Using "en_TW" translation
12-18 19:36:47:668 [ info gui.socketapi ]:	server started, listening at  "com.owncloud.desktopclient.socketApi"
12-18 19:36:47:668 [ info gui.folder.manager ]:	setting remote poll timer interval to 30000 msec
12-18 19:36:47:917 [ info gui.application ]:	Tray menu workarounds: noabouttoshow: true fakedoubleclick: false showhide: false manualvisibility: true
12-18 19:36:47:938 [ info gui.folder.manager ]:	Setup folders from  "/Users/yen/Library/Preferences/ownCloud/folders" (migration)
12-18 19:36:47:939 [ info sync.clientproxy ]:	Set proxy configuration to use system configuration
12-18 19:36:47:999 [ info gui.application ]:	No configured folders yet, starting setup wizard
12-18 19:36:50:801 [ info gui.updater ]:	Checking for available update
12-18 19:36:50:801 [ info sync.accessmanager ]:	2 "" "https://updates.owncloud.com/client/?version=2.5.1.10818&platform=macos&oem=ownCloud&versionsuffix=" has X-Request-ID "14d56fb0-00c4-49c3-90d4-403c448c27bf"
12-18 19:36:52:081 [ info gui.updater ]:	Client is on latest version!
```

2. Web server error log: N/A

3. Server logfile: ownCloud log (data/owncloud.log): N/A

### PR DESCRIPTION
**Pull request by ckamm**
The updated code in qtmacgoodies uses private headers.

### CODE REVIEW

**Comment by dschmidt** [comment]
It's probably not worth updating qtmacgoodies again and then to update this PR, but for the record the "modern CMake" approach would have been to link against `Qt5::GuiPrivate` imported target.

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -174,6 +174,9 @@ if (APPLE)
         ../3rdparty/qtmacgoodies/src/macstandardicon.mm
         ../3rdparty/qtmacgoodies/src/macwindow.mm
         )
+    # We want to access Cocoa specific structures in the code above
+    # and need the platform plugin interface for that - which is private.
+    include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
```
Comment (dschmidt) [inline src/gui/CMakeLists.txt]: I would suggest to use ``target_include_directories(${APPLICATION_EXECUTABLE} ${Qt5Gui_PRIVATE_INCLUDE_DIRS})`` or even better  ``target_link_libraries(${APPLICATION_EXECUTABLE} Qt5::GuiPrivate)``instead.

Comment (yan12125) [reply src/gui/CMakeLists.txt]: Thanks for the update! Specifying the target does not work here - cmake complains `Cannot specify link libraries for target "owncloud" which is not built by this project.` ( or `include directories` for the first suggestion by @dschmidt. I get it work by modifying existing `target_link_libraries` command:

```Diff
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 9eaf326f8..45de10b2e 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -272,7 +272,7 @@ set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY}
 )

-target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5::Widgets Qt5::Network Qt5::Xml)
+target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5::GuiPrivate Qt5::Widgets Qt5::Network Qt5::Xml)
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
```
And of course, there's no crash with the patched app.

**Comment by ckamm** [comment]
Still needs that Qt5::GuiPrivate link library target, right?

**Comment by yan12125** [comment]
> PR updated

Don't have my Mac now. Will test the updated version tomorrow morning (GMT+8)

**Pull request merged**